pkg/predict/input.go

```diff
@@ -55,19 +59,54 @@ func NewInputsWithBaseDir(keyVals map[string]string, baseDir string) Inputs {
 	return input
 }
 
-func (inputs *Inputs) toMap() (map[string]string, error) {
-	keyVals := map[string]string{}
+func (inputs *Inputs) toMap() (map[string]interface{}, error) {
+	keyVals := map[string]interface{}{}
 	for key, input := range *inputs {
 		if input.String != nil {
+			// Directly assign the string value
 			keyVals[key] = *input.String
 		} else if input.File != nil {
-			content, err := os.ReadFile(*input.File)
+			// Single file handling: read content and convert to a data URL
+			dataURL, err := fileToDataURL(*input.File)
 			if err != nil {
 				return keyVals, err
 			}
-			mimeType := mime.TypeByExtension(filepath.Ext(*input.File))
-			keyVals[key] = dataurl.New(content, mimeType).String()
+			keyVals[key] = dataURL
+		} else if input.Array != nil {
+			// Handle array, potentially containing file paths
+			var dataURLs []string
+			for _, elem := range *input.Array {
+				if str, ok := elem.(string); ok && strings.HasPrefix(str, "@") {
+					filePath := str[1:] // Remove '@' prefix
+					dataURL, err := fileToDataURL(filePath)
+					if err != nil {
+						return keyVals, err
+					}
+					dataURLs = append(dataURLs, dataURL)
+				} else if ok {
+					// Directly use the string if it's not a file path
+					dataURLs = append(dataURLs, str)
+				}
+			}
+			keyVals[key] = dataURLs
```

Comment:
It's not a big deal, but it would be more conventional when you know the length of the slice beforehand to do something like

```suggestion
			dataURLs := make([]string, len(input.Array))
			for i, elem := range input.Array {
				if str, ok := elem.(string); ok && strings.HasPrefix(str, "@") {
					filePath := str[1:] // Remove '@' prefix
					dataURL, err := fileToDataURL(filePath)
					if err != nil {
						return keyVals, err
					}
					dataURLs[i] = dataURL
				} else if ok {
					// Directly use the string if it's not a file path
					dataURLs[i] = str
				}
			}
			keyVals[key] = dataURLs
```